rfd/0086/mariposa.md

```diff
@@ -0,0 +1,96 @@
+## Mariposa integration
+
+There are a number of cases where we've wanted for lifecycle hooks to be able to communicate information back to ContainerPilot, or to be otherwise able to alter the ContainerPilot environment for purposes of future handler events or the main application itself. Additionally, it's likely that we at Joyent will want the [RFD36](https://github.com/joyent/rfd/blob/master/rfd/0036/README.md) scheduler for Triton to be able to have an interface to communicate with containers and ContainerPilot itself, for which we can provide an example in ContainerPilot.
+
+This control plane would be useful for event hooks as well. Telemetry sensor outputs will be consumed via a robust API instead of the existing brittle text scraping. Event hooks will also be able to set environment variables for other services and event hooks.
+
+The proposal is in two parts. A read-only endpoint exposing the state of all services associated with this ContainerPilot instance will be exposed as a new endpoint in the Telemetry HTTP server. A second endpoint that accepts HTTP POST updates to the state will be implemented as a listener on a unix domain socket only (at least in the initial implementation).
+
+### Telemetry endpoint
+
+##### `GetStatus GET /status`
+
+This API will expose the state of all services associated with this ContainerPilot instance.
+
+*Example Response*
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+Content-Length: 328
+{
+  "services": [
+    {
+      "name": "nginx",
+      "address": "192.168.1.100",
+      "port": 80,
+      "status": "healthy"
```

Comment:
Mariposa also assumes ContainerPilot can share what upstreams this service depends on.